### PR TITLE
Add logcat crash parsing and reporting

### PR DESCRIPTION
**Pull request by lukaville**
### Stacktrace message for native crashes

Before:
image

After:
image

### Stacktrace message for JVM crashes

Crash outside of test run, for example in Application.onCreate

Before:
```
Instrumentation run failed due to 'Process crashed.'
```

After:
```
Crash events:
* FATAL EXCEPTION: main
Process: com.example, PID: 12571
java.lang.RuntimeException: Unable to create application com.example.TestApplication: java.lang.RuntimeException: Qwerty
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6465)
	at android.app.ActivityThread.access$1300(ActivityThread.java:219)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
Caused by: java.lang.RuntimeException: Qwerty
	at com.example.TestApplication.onCreate(Application.kt:10)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1182)
	at android.support.test.runner.MonitoringInstrumentation.callApplicationOnCreate(MonitoringInstrumentation.java:382)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6460)
	... 8 more

Instrumentation run failed due to 'Process crashed.
```

### CODE REVIEW

File: html-report/src/components/ColorizedLogsComponent.js [hunk 1]
```diff
@@ -56,58 +57,116 @@ export default class ColorizedLogsComponent extends Component {
...
+ *
+ * <pre>04-06 16:21:43.746 1852-25383/? D/GraphicBufferSource: got buffer with new dataSpace #104</pre>
+ *
+ * Group 1: Date + Time
+ * Group 2: PID
+ * Group 3: TID (hex on some systems!)
+ * Group 4: Log Level character
+ * Group 5: Tag
+ */
+function _parseLogcatMessages(response) {
```
Comment (lukaville) [inline html-report/src/components/ColorizedLogsComponent.js:117]: updated parsing so it supports messages with one header and multiline message body.
previously each line of the body was with its own header (so it wasn't possible to copy the whole stack trace easily)

Comment (lukaville) [reply html-report/src/components/ColorizedLogsComponent.js:117]: image

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/logcat/LogcatCollector.kt [hunk 2]
```diff
@@ -79,6 +94,11 @@ class LogcatCollector : LogcatEventsListener, LogReportProvider {
 
     private data class DeviceState(
         val currentBatchId: String? = null,
-        val currentTest: LogTest? = null
+        val currentTest: TestState? = null
+    )
+
+    private data class TestState(
+        val test: LogTest,
+        val processId: Int
```
Comment (lukaville) [inline vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/logcat/LogcatCollector.kt:102]: add process id for test state to filter crashes related only to the test execution

File: vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/shell/LogcatParserListener.kt [hunk 3]
```diff
@@ -1,21 +1,48 @@
...
 class LogcatParserListener(
     private val device: IDevice,
     private val receiver: (List<LogCatMessage>) -> Unit
 ) : TailerListenerAdapter() {
 
     private val parser = LogCatLongEpochMessageParser()
+    private val messageBuffer = StringBuilder()
+    private var lastHeader: LogCatHeader? = null
 
     override fun handle(line: String?) {
```
Comment (lukaville) [inline vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/shell/LogcatParserListener.kt:22]: updated logic of the parser so now it merges all the lines with the same header (previously crash stacktrace was reported in separate logcat messages but with the same header)

**Review by yufimtsev — APPROVED** [review]

**Pull request merged**